File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> umap;
    string answer = "";
    for(auto a : participant){
        if(umap.end() == umap.find(a)){
             umap.insert(make_pair(a, 1));
        }else{
            umap[a]++;
        }
    }
    for (auto a : completion)
        umap[a]--;

    for(auto a : participant)
        if (umap[a] > 0)
        {
            answer = a;
            break;
        }

    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion){
        Map<String, Integer> remainMap = new HashMap<>();
        for (String s : participant) {
            remainMap.put(s, remainMap.getOrDefault(s, 0) + 1);
        }
        for (String s : completion) {
            remainMap.put(s, remainMap.get(s) - 1);
        }
        String answer = "";
        for (String s : remainMap.keySet()) {
            if (remainMap.get(s) != 0) 
                answer = s;
        }
        return answer;
    }
}